(GH-73) (GH-72) (GH-70) (GH-69) feat: Login function and updated user routes, test, and index file

Alessandro-Tiseo opened the pull request:
# Objectives of the Pull Request ? 
> Describe the high level purpose of your pull request. What are you trying to achieve ? How are you doing it ?

Added the login function to the file called userRoutes.js. Change/Added a few lines of code in the file called server.js. Added the index.html to test the physical login. Created two files called listen.js and function.js. listen.js contains code to test the physical login. function.js has a function to randomly generate passwords for web. In addition, added test cases to test login function. Lastly, added a function to create passwords for non-web which randomly generate passwords for the test cases.


# What is left out of the Pull Request ? 
> Describe what is not included in the pull request. Why did you not include it in the PR. What are the next steps ?
The pull request is missing a detailed UI for the login and signup screens within the app frame. Currently, a basic index.html file is used as a placeholder for the features of sign up and log in. 


# How to run the pull request ?
> Provide the code required to run the pull request. This is the code that will be used to review your pull request. **The provided code must work as-is. If a react-native error is raised while running the code, the PR will be rejected. The following code / placeholder is only provided as documentation / helper to get you started and you will need to adjust it.**

```bash
# Assuming you are in a folder containing minicap_concordia_campus_guide_app/application folder.
# Update the code
git checkout feature/loginAuthentication
git pull
# Might be required if you update check the PR

# Run npx
npx expo start
```
To run test cases use: npm test
To run code with the html file:
1 - Change directory to the backend folder.
2 - Enter this command inside the terminal: node server.js 
3 - To visit the login function go to http://localhost:3000/index.html
Note: Change port number if it does not work. 
# Checklist
> Please, read carefully each item before checking it. Your PR's review might be delayed otherwise.

* **Code** :
  * [x] I have added all necessary unit tests to demonstrate that my code functions as expected and is testable.
  * [x] The code I'm asking a review for is working. **I understand that my PR will be rejected as-is otherwise.**
  * [x] My code follows the conventions described in the `contribution guide`.
  
![E6CB0C17-8C83-4118-9BE9-35ABE9565F72](https://github.com/user-attachments/assets/0d5ecd59-4c53-4074-9802-f842b06ae23e)

 
* **Documentation** : 
  * [x]  I have updated the documentation (README) accordingly to my changes.
 
* **Pull-request** : 
  * [x]  My merge message is prefixed with a `commitizen` key word so it looks something like `<feat|fix|chore|doc|refactor|perf|style>: foo bar`**
  * [x]  The code *provided to run the pull request* is working. **I understand that my PR will be rejected as-is otherwise.**
  * [x]  My pull request is documented. I have explained the needs for the PR, what was left out of the it and why.
  * [x]  I have carefully reviewd each changes made to a file and made sure the files included on the PR were actually added on purpose.
  * [x]  I have assignee at least **1** people to review my PR.

Alessandro-Tiseo committed:
Login function and updated user routes, test, and index file

AlimuratDinch commented:
@Alessandro-Tiseo the Pull Request name should be like `(GH-73) (GH-72) (GH-69) feat: login function and updated user routes, test, and index file `

Alessandro-Tiseo commented:
@[AlimuratDinch](https://github.com/AlimuratDinch) I changed the name of the pull request to (GH-73) (GH-72) (GH-69) feat: Login function and updated user routes, test, and index file  .

Sawanoza commented:
1. Make sure to check your boxes in the checklist when writing a PR
2. The backend folder doesn't follow the structure of our repo, it should be inside the application folder instead of floating between application and documents
3. SonarQube has brought up 3 security hotspots. Instead of having hardcoded values, you could use a randomly generated number for passwords. 
4. For Code Quality, I believe readability can be enhanced by adding relevant comments to certain methods instead of using them only as delimiters. Especially since another person will be integrating this login feature to match with the UI design.

Sawanoza reviewed (changes requested):
@Alessandro-Tiseo 
1. Make sure to check your boxes in the checklist when writing a PR
2. Refactoring: The backend folder doesn't follow the structure of our repo, it should be inside the application folder instead of floating between application and documents
3. SonarQube has brought up 3 security hotspots. Instead of having hardcoded values, you could use a randomly generated number for passwords. 
4. For Code Quality, I believe readability can be enhanced by adding relevant comments to certain methods instead of using them only as delimiters. Especially since another person will be integrating this login feature to match with the UI design.

Alessandro-Tiseo committed:
Fixing The Password SonarQube

Alessandro-Tiseo committed:
Made passwords more secure

Alessandro-Tiseo commented:
* [x]  Checked the boxes in the checklist when writing a PR. 
* [x]  Fix SonarQube security hotspots.

NekZampe reviewed (changes requested):
1. Please edit the run instructions to indicate to go the backend folder not the application folder.
2. Also there's frontend HTML code added to the backend which shouldn't be there. I suggest you use a free third party software like postman to manually test your backend routes while developing.

Other than that looks good!

Sawanoza reviewed (approved):
Changes requested have been done. Only missing code quality improvements by adding relevant comments to the tests, so the integration with the frontend is easier to complete (Can be done in parallel with the frontend). Also, will need a refactoring PR to merge the backend folder with the application folder once we sort it out.

NekZampe commented:
> Changes requested have been done. Only missing code quality improvements by adding relevant comments to the tests, so the integration with the frontend is easier to complete (Can be done in parallel with the frontend). Also, will need a refactoring PR to merge the backend folder with the application folder once we sort it out.

Backend should not be merged with the application folder. The Application is the mobile app code whereas the backend is for server side code such as creating users, rendering indoor routes, etc.

Alessandro-Tiseo committed:
Updated branch
